src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionParameterNumberTest.java

```diff
@@ -0,0 +1,106 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2024 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package org.checkstyle.suppressionxpathfilter;
+
+import static com.puppycrawl.tools.checkstyle.checks.sizes.ParameterNumberCheck.MSG_KEY;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.checks.sizes.ParameterNumberCheck;
+
+public class XpathRegressionParameterNumberTest extends AbstractXpathTestSupport {
+
+    @Override
+    protected String getCheckName() {
+        return ParameterNumberCheck.class.getSimpleName();
+    }
+
+    @Test
+    public void testDefault() throws Exception {
+        final File fileToProcess =
+                new File(getPath("SuppressionXpathRegressionParameterNumberDefault.java"));
+
+        final DefaultConfiguration moduleConfig = createModuleConfig(ParameterNumberCheck.class);
+
+        final String[] expectedViolations = {
+            "5:10: " + getCheckMessage(ParameterNumberCheck.class, MSG_KEY, 7, 11),
+        };
+
+        final List<String> expectedXpathQueries = Collections.singletonList(
+                "/COMPILATION_UNIT/CLASS_DEF"
+                + "[./IDENT[@text='SuppressionXpathRegressionParameterNumberDefault']]"
+                + "/OBJBLOCK/METHOD_DEF/IDENT[@text='myMethod']"
+        );
+
+        runVerifications(moduleConfig, fileToProcess, expectedViolations, expectedXpathQueries);
+
+    }
+
+    @Test
+    public void testMethods() throws Exception {
+        final File fileToProcess =
+                new File(getPath("SuppressionXpathRegressionParameterNumberMethods.java"));
+
+        final DefaultConfiguration moduleConfig = createModuleConfig(ParameterNumberCheck.class);
+        moduleConfig.addAttribute("max", "10");
```

Comment:
CI is not happy of this usage
https://app.circleci.com/pipelines/github/checkstyle/checkstyle/23705/workflows/ec082a1a-878e-429b-a0bd-a602f97a6c95/jobs/510223/artifacts
please recheck how others do not use this deprecation.

Reply:
@romani done. used addProperty instead